src/axom/core/StaticArray.hpp

```diff
@@ -48,24 +48,10 @@ class StaticArray : public StackArray<T, N>
    */
   AXOM_HOST_DEVICE StaticArray(const StaticArray &obj) : StackArray<T, N>(obj), m_size(obj.m_size)
   {
-    for(int i = 0; i < obj.m_size; i++)
+    for(axom::IndexType i = 0; i < obj.m_size; i++)
     {
       StackArray<T, N>::m_data[i] = obj.StackArray<T, N>::m_data[i];
     }
-    m_size = obj.m_size;
-  }
-
-  /*!
-   * \brief Move Constructor
-   * \param obj The object to be moved.
-   */
-  AXOM_HOST_DEVICE StaticArray(StaticArray &&obj) : StackArray<T, N>(obj), m_size(obj.m_size)
-  {
-    for(int i = 0; i < obj.m_size; i++)
-    {
-      StackArray<T, N>::m_data[i] = obj.StackArray<T, N>::m_data[i];
-    }
-    m_size = obj.m_size;
   }
```

Comment:
Was there a reason for removing the move constructor/assignment operator?

Reply:
There was an objection to having them in a [previous PR](https://github.com/LLNL/axom/pull/1523#pullrequestreview-2737127956) so in this PR I removed them.